lib/ansible/plugins/inventory/docker_swarm.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018, Stefan Heitmueller <stefan.heitmueller@gmx.com>
+# Copyright (c) 2018 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: docker_swarm
+    plugin_type: inventory
+    authors:
+      - Stefan Heitmüller (stefan.heitmueller@gmx.com)
+    short_description: Ansible dynamic inventory plugin for Docker swarm nodes.
+    requirements:
+        - python >= 2.7
+        - Docker SDK for Python > 1.10.0
+    description:
+        - Reads inventories from the Docker swarm API.
+        - Uses a YAML configuration file docker_swarm.[yml|yaml].
+    options:
+        plugin:
+            description: the name of this plugin, it should always be set to 'docker_swarm' for this plugin to recognize it as it's own.
+            required: true
+            choices: ['docker_swarm']
+        host:
+            description: Socket of a Docker swarm manager node (tcp,unix).
+            required: true
+'''
```

Comment:
```suggestion
        tls:
            description: Use TLS
            required: false
        tls_verify:
            description: TODO
            required: false
        key_path:
            description: TODO
            required: false
        cacert_path:
            description: TODO
            required: false
        cert_path:
            description: TODO
            required: false
        ssl_version:
            description: TODO
            required: false
        tls_hostname:
            description: TODO
            required: false
'''
```

Part of the code for TLS support